### PR TITLE
#1642 & #1559

### PR DESCRIPTION
**Pull request by salabh-aot**
- #1642
     - Updated existing staff entries email to lower case
     - Refactored code to set email to lower case when adding/updating staffs
 - #1559
     - Fixed the issue where task template upload was failing when `tips` column was empty

### CODE REVIEW

File: epictrack-api/src/api/services/staff.py [hunk 1]
```diff
@@ -96,12 +96,12 @@ def find_by_id(cls, _id):
...
     def check_existence(cls, email, staff_id=None):
         """Checks if a staff exists with given email address"""
-        return Staff.check_existence(email, staff_id)
+        return Staff.check_existence(email.lower(), staff_id)
 
     @classmethod
     def find_by_email(cls, email):
         """Find staff by email address"""
-        return Staff.find_by_email(email)
+        return Staff.find_by_email(email.lower())
```
Comment (dinesh-aot) [inline epictrack-api/src/api/services/staff.py]: Is it good to have this convertion at the schema level like the way you did it in the POST

File: epictrack-api/src/api/schemas/request/work_request.py [hunk 2]
```diff
@@ -192,7 +192,6 @@ class WorkStatusParameterSchema(RequestBodyParameterSchema):
 
     description = fields.Str(
         metadata={"description": "description of status"},
-        validate=validate.Length(max=500),
         required=True,
```
Comment (dinesh-aot) [inline epictrack-api/src/api/schemas/request/work_request.py:196]: Keep the validation there. We don't want to remove it. We are waiting to hear from the ops team their requirements

File: epictrack-web/src/components/workPlan/issues/IssuesForm.tsx [hunk 3]
```diff
@@ -141,19 +145,11 @@ const IssuesForm = () => {
           />
         </Grid>
         <Grid item xs={12}>
-          <ETFormLabelWithCharacterLimit
-            characterCount={watchedDescription.length}
-            maxCharacterLength={descriptionCharacterLimit}
-          >
-            Description
-          </ETFormLabelWithCharacterLimit>
+          <ETFormLabel>Description</ETFormLabel>
```
Comment (dinesh-aot) [inline epictrack-web/src/components/workPlan/issues/IssuesForm.tsx]: Keep the character count stuff there as I mentioned before

File: epictrack-web/src/components/workPlan/status/StatusForm.tsx [hunk 4]
```diff
@@ -93,22 +92,13 @@ const StatusForm = () => {
...
         </Grid>
         <Grid item xs={12}>
-          <ETFormLabelWithCharacterLimit
-            characterCount={description.length}
-            maxCharacterLength={CHARACTER_LIMIT}
-            required
-          >
-            Description
-          </ETFormLabelWithCharacterLimit>
+          <ETFormLabel>Description</ETFormLabel>
```
Comment (dinesh-aot) [inline epictrack-web/src/components/workPlan/status/StatusForm.tsx]: Keep the character count and other restrictions there as I mentioned above

File: epictrack-api/src/api/schemas/request/staff_request.py [hunk 5]
```diff
@@ -88,10 +95,15 @@ class StaffBodyParameterSchema(RequestBodyParameterSchema):
...
-        metadata={"description": "Active status of the staff"},
-        required=True
+        metadata={"description": "Active status of the staff"}, required=True
     )
 
+    @pre_load
+    def convert_email_to_lower(self, data, **kwargs):  # pylint: disable=unused-argument
+        """Converts staff email into lower case string"""
+        data["email"] = data["email"].lower()
+        return data
```
Comment (saravanpa-aot) [inline epictrack-api/src/api/schemas/request/staff_request.py]: safer to check if email exists
 if 'email' in data:

I know email might be mandatory here , but that one day when people makes the email optional and foorgot to change this methiod , we will have the bug..so as a better practice

**Pull request merged**